lib/liquid/filters.ex

```diff
@@ -80,4 +448,24 @@ defmodule Liquid.Filters do
     filter(rest, ret)
   end
 
+  def filter(a, b, _) when a == [] do
+    filter(a, b)
+  end
+
+  def filter([head|tail], b, context) do
+    case context.assigns do
```

Comment:
Well the argument value is an unparsed value. That still needs to be parsed to either the primitive or the result of the variable.

But with the thing we talked about on slack might be better parsing all the arguments already in Liquid.Variable before passing it to the filter.